services/app/src/contexts/ChessBoardContext.jsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable react/jsx-no-constructed-context-values */
+/* eslint-disable react/prop-types */
+import React, { createContext, useState } from 'react';
+import { useChessBoard } from '../hooks/useChessBoard';
+import { NUMBER_OF_ROWS, NUMBER_OF_COLUMNS } from '../constants/board';
+import '../static/style/gameScore.css';
+
+export const ChessBoardContext = createContext();
+let num;
+
+const ChessBoardContextProvider = ({ children }) => {
+  const [selectedItem, setSelectedItem] = useState('');
+  const [board, setBoard] = useChessBoard();
+
+  const placeItemOnBoard = (row, tile) => {
+    if (selectedItem) {
+      const copyOfBoard = [...board];
+      copyOfBoard[row][tile].items.push(
+        selectedItem,
+        calculateTheScore(row, tile, selectedItem),
+        // + calculateTheScore(row, tile, selectedItem),
+      );
+      setBoard(copyOfBoard);
+      setSelectedItem('');
+    }
+  };
+
+  const calculateTheScore = (row, tile, selectedItem) => {
+    var selectedX = 0;
+    var selectedY = 0;
+    var selectedPiece = 0;
+
+    const xValues = new Map(
+      Object.entries({
+        1: 1,
+        2: 2,
+        3: 3,
+        4: 5,
+        5: 8,
+        6: 13,
+      }),
+    );
+    const yValues = new Map(
+      Object.entries({
+        1: 1,
+        2: 2,
+        3: 3,
+        4: 5,
+        5: 8,
+        6: 13,
+      }),
+    );
+
+    const pieces = new Map(
+      Object.entries({
+        pawn: 1,
+        knight: 2,
+        bishop: 3,
+        rook: 5,
+        king: 8,
+        queen: 13,
+      }),
+    );
+
+    for (const [key, value] of xValues.entries()) {
```

Comment:
To make it more performant and concise, you can replace `Map`s with plain objects like this:
```
    const xValueToScoreMap = {
        1: 1,
        2: 2,
        3: 3,
        4: 5,
        5: 8,
        6: 13,
    };
```
and the use those objects when calculating scores
```
  const xScore = xValueToScoreMap[row] ?? 0;
  const yScore = yValueToScoreMap[column] ?? 0;
  const pieceScore = pieceToScoreMap[selectedPiece] ?? 0;
```